### PR TITLE
Add a status field to `listpeers` indicating where an HTLC is currently being held

### PR DESCRIPTION
**Pull request by cdecker**
Debugging where an HTLC is currently being held can help us identify
plugins that are not behaving correctly. On top of the pure `status`
field in the `htlcs` array we also print debug messages when entering
and leaving a plugin hook, so logs are a bit more helpful too.

### CODE REVIEW

File: lightningd/htlc_end.h [hunk 1]
```diff
@@ -55,6 +55,9 @@ struct htlc_in {
 	struct secret blinding_ss;
 	/* true if we supplied the preimage */
 	bool *we_filled;
+
+	/* A simple text annotation shown in `listpeers` */
+	char *status;
```
Comment (rustyrussell) [inline lightningd/htlc_end.h:60]: This seems to be in the wrong commit?

File: lightningd/peer_htlcs.c [hunk 2]
```diff
@@ -1015,8 +1015,14 @@ static void htlc_accepted_hook_serialize(struct htlc_accepted_hook_payload *p,
...
 	const struct route_step *rs = p->route_step;
-	const struct htlc_in *hin = p->hin;
+	struct htlc_in *hin = p->hin;
 	s32 expiry = hin->cltv_expiry, blockheight = p->ld->topology->tip->height;
+
+	tal_free(hin->status);
+	hin->status =
+	    tal_fmt(hin, "Waiting for the htlc_accepted hook of plugin %s",
+		    plugin->shortname);
+
```
Comment (rustyrussell) [inline lightningd/peer_htlcs.c:1025]: You presumably should initialize this to NULL everywhere hin is allocated?

**Review by rustyrussell — COMMENTED** [review]
Seems some uninitialized fields, should be easy to fix?

**Comment by cdecker** [comment]
Got my commits mixed up it seems, I'll reslice them and update here.

**Comment by rustyrussell** [comment]
Missing regeneration of files, fixed (and trivial rebase on master).

Ack 5dc4b55706619e3d15a07e91e15a66768955dc67

**Comment by rustyrussell** [comment]
Oops, and you freed it without setting to NULL, causing a use-after-free.

```
diff --git a/lightningd/peer_htlcs.c b/lightningd/peer_htlcs.c
index 65a704023..d0887d650 100644
--- a/lightningd/peer_htlcs.c
+++ b/lightningd/peer_htlcs.c
@@ -1074,7 +1074,7 @@ htlc_accepted_hook_final(struct htlc_accepted_hook_payload *request STEALS)
 	struct htlc_in *hin = request->hin;
 	struct channel *channel = request->channel;
 
-	tal_free(request->hin->status);
+	request->hin->status = tal_free(request->hin->status);
 
 	/* *Now* we barf if it failed to decode */
 	if (!request->payload) {
```

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/commit/b89ce785b4969ae6a3fcdd3d6c93e1823a73d508

**Pull request merged**